script/model/forecast.py

```diff
@@ -0,0 +1,520 @@
+    oltp_server = TestOLTPBench(DEFAULT_OLTP_SERVER_ARGS)
+    db_server = oltp_server.db_instance
+    db_server.run_db()
+
+    # Download the OLTP repo and build it
+    oltp_server.run_pre_suite()
+
+    # Load the workload pattern - based on the tpcc.json in
+    # testing/oltpbench/config
+    test_case_config = DEFAULT_OLTP_TEST_CASE
+    test_case_config["weights"] = tpcc_weight
+    test_case = TestCaseOLTPBench(test_case_config)
+
+    # Prep the test case build the result dir
+    test_case.run_pre_test()
+
+    rates = tpcc_rates * pattern_iter
+    gen_forecast_data(test_case.xml_config, rates)
+
+    # Run the actual test
+    ret_val, _, stderr = run_command(test_case.test_command,
+                                     test_case.test_error_msg,
+                                     cwd=test_case.test_command_cwd)
+    if ret_val != ErrorCode.SUCCESS:
+        LOG.error(stderr)
+        return False
+
+    # Clean up, disconnect the DB
+    db_server.stop_db()
+    db_server.delete_wal()
+
+    if not Path(DEFAULT_QUERY_TRACE_FILE).exists():
+        LOG.error(
+            f"Missing {DEFAULT_QUERY_TRACE_FILE} at CWD after running OLTP TPCC")
+        return False
+
+    return True
+
+
+class DataLoader:
+    # Hardcoded timestamp column index in the query_trace file
+    TS_IDX = 1
+
+    def __init__(
+            self,
+            query_trace_file: str = DEFAULT_QUERY_TRACE_FILE,
+            test: bool = False,
+            eval_size: int = EVAL_DATA_SIZE,
+            seq_len: int = SEQ_LEN,
+            horizon_len: int = HORIZON_LEN) -> None:
+        """
+        :param query_trace_file: Where to process the query trace
+        :param test: If the Loader is for testing
+        :param eval_size:  Number of data points used for evaluation(testing)
+        :param seq_len: Length of a sequence
+        :param horizon_len: Horizon length
+        """
+        self.query_trace_file = query_trace_file
+
+        self.seq_len = seq_len
+        self.horizon_len = horizon_len
+        self.scaler = MinMaxScaler(feature_range=(-1, 1))
+        self.test_mode = test
+        self.eval_data_size = eval_size
+
+        self._load_data()
+
+    def _load_data(self) -> None:
+        LOG.info(f"Loading data from {self.query_trace_file}")
+        # Load data from the files
+        with open(self.query_trace_file, newline='') as csvfile:
+            reader = csv.DictReader(csvfile)
+            data = np.array(
+                [[int(r['query_id']), int(r[' timestamp'])] for r in reader])
+
+            data = self._aggregate_data(data, INTERVAL_MS)
+            self.train_raw_data, self.test_raw_data = self._split_data(data)
+            LOG.info(
+                f"Loaded data successfully: num_test={len(self.test_raw_data)}, num_train={len(self.train_raw_data)}, "
+                f"seq_len={self.seq_len}, horizon={self.horizon_len}")
+
+    def _normalize_data(self, data: np.ndarray) -> np.ndarray:
```

Comment:
`_normalize_data` should belong to the `Model` class, as discussed below. Again, look at `script/model/model.py` for reference.